sdks/besu/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -0,0 +1,222 @@
+    assetExchangeAgreement.setRecipient(recipient);
+    assetExchangeAgreement.setLocker(locker);
+    return Buffer.from(assetExchangeAgreement.serializeBinary())
+}
+
+
+function createHybridAssetExchangeAgreementSerialized(assetType: string, assetID: string, assetData: string, numUnits: number, recipient: string) {
+    const assetExchangeAgreement = new assetLocksPb.HybridAssetExchangeAgreement();
+    assetExchangeAgreement.setAssettype(assetType);
+    assetExchangeAgreement.setId(assetID);
+    assetExchangeAgreement.setRecipient(recipient);
+    assetExchangeAgreement.setAssetdata(Buffer.from(assetData, "hex"));
+    assetExchangeAgreement.setNumunits(numUnits);
+    return Buffer.from(assetExchangeAgreement.serializeBinary())
+}
+
+// Create an asset lock structure
+function createAssetLockInfoSerialized(hash: Hash, expiryTimeSecs: number) {
+    const lockInfoHTLC = new assetLocksPb.AssetLockHTLC();
+    lockInfoHTLC.setHashmechanism(hash.HASH_MECHANISM);
+    lockInfoHTLC.setHashbase64(Buffer.from(hash.getSerializedHashBase64()));
+    lockInfoHTLC.setExpirytimesecs(expiryTimeSecs);
+    lockInfoHTLC.setTimespec(assetLocksPb.TimeSpec.EPOCH)
+    return Buffer.from(lockInfoHTLC.serializeBinary())
+}
+
+// Create an asset claim structure
+function createAssetClaimInfoSerialized(hash: Hash) {
+    const claimInfoHTLC = new assetLocksPb.AssetClaimHTLC();
+    claimInfoHTLC.setHashmechanism(hash.HASH_MECHANISM);
+    claimInfoHTLC.setHashpreimagebase64(Buffer.from(hash.getSerializedHashBase64()));
+    const claimInfoHTLCSerialized = claimInfoHTLC.serializeBinary();
+    const claimInfo = new assetLocksPb.AssetClaim();
+    claimInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    claimInfo.setClaiminfo(claimInfoHTLCSerialized);
+    return Buffer.from(claimInfo.serializeBinary())
+}
+
+const createHTLC = async (
+    assetManagerContract: any,
+    tokenContract: any,
+    assetType: string,
+    assetID: string,
+    assetData: string,
+    assetAmount: number,
+    senderAddress: string,
+    recipientAddress: string,
+    expiryTimeSecs: number,
+    hash: Hash
+): Promise<{ result: any }> => {
+
+    if (!assetManagerContract) {
+        console.log("Contract handle not supplied");
+        return { result: false };
+    }
+    if (!recipientAddress) {
+        console.log(`Recipient address not supplied ${recipientAddress}`);
+        return { result: false };
+    }
+    const currTimeSecs = Math.floor(Date.now() / 1000);   // Convert epoch milliseconds to seconds
+    if (expiryTimeSecs <= currTimeSecs) {
+        console.log("Supplied expiry time invalid or in the past: %s; current time: %s", new Date(expiryTimeSecs).toISOString(), new Date(currTimeSecs).toISOString());
+        return { result: false };
+    }
+    var protobufParams;
+    if (assetType == "ERC721") {
+        protobufParams = createAssetExchangeAgreementSerialized("", String(assetID), recipientAddress.slice(2), Web3.utf8ToHex((assetData)));
```

Comment:
Instead of if-else like this can you add 3 functions? createFungibleHTLC, createNonFungibleHTLC, createHTLC, similar to `lockFungibleAsset`, `lockNonFungibleAsset`, and `lockAsset` in contract?